dovecot/dovecot-lib.pl

```diff
@@ -11,6 +11,11 @@
 @mail_envs = ( undef, "maildir:~/Maildir", "mbox:~/mail/:INBOX=/var/mail/%u",
 	       "maildir:~/Maildir:mbox:~/mail/" );
 
+# Dovecot version specific mapping if any
+our $base_version = &get_dovecot_based_version();
+require "$module_root_directory/dovecot$base_version-lib.pl"
+	if (-r "$module_root_directory/dovecot$base_version-lib.pl");
```

Comment:
Instead of creating a complex method of incuding a file based on the Dovecot version and the overhead of querying the version every time dovecot-lib.pl is loaded, why not just make the `map_find` function check the version and return a different hash?